docusaurus/docs/iOS/client/controllers/channels.md

```diff
@@ -55,49 +55,73 @@ let compoundFilter: Filter<ChannelListFilterScope> = .and([
 The `query.sort` is an array of sorting options. Sorting options are applied based on their order in the array so the first option has the highest impact while the others are used mainly as a tiebreakers. By default, the channel list is sorted by `updated_at`.
 
 ### 2. Create a controller
-#### With auto Channel Filtering
-The controller can be created using `ChatClient`:
+The simplest way to create a controller is by using the static method `channelListController(query:)` on your `ChatClient`.
 ```swift
 let controller = ChatClient.shared.channelListController(query: query)
 ```
-The behaviour is controlled from the `ChatClientConfig` and more specifically the `isChannelAutomaticFilteringEnabled` flag. 
+By default, the SDK will automatically handle filtering the channels as they get created. Whenever there is a web socket event that a channel has been created, the SDK will only insert it in the channel list if it matches the query.
 
-When the flag is enabled (`true`), then the new/updated channels are being filtered automatically, by matching the `query.filter`.
-By default, the SDK will handle filtering the channels as they get updated and will only return to the controller **only** the channels that match the provider `query.filter`. 
-This means, that the controller always receives channels that match its `query.filter` and then the controller is responsible for deciding if the channels need to be linked/unlinked.
+:::note
+In cases though, where the query provided contains, extra data or custom filters, **the SDK may not be able to automatically match the filter query**. In this case, you will need to provide a filtering closure.
+:::
 
-When the flag is disabled (`false`), then the new/updated channels won't be filtered as the arrive, and the controller will be responsible for filtering them first and confirm that they
-match its `query.filter` and then decide if the filtered channels need to be linked/unlinked.
+#### Filtering with extra data
+First we need to disable the Channel auto-filtering. We can do that by turning the `isChannelAutomaticFilteringEnabled` in your `ChatClient` configuration, to `false`.
```

Comment:
Not sure if this should be the suggested approach. We can also suggest an "additive" block, right?

Reply:
You are right, Nuno had a similar idea but I have to admit that i was a little bit confused. I have now updated the docs with the `filter closure` being described as additive. Additionally I added a test that demonstrates the scenario where a custom filter is applied and the auto-filtering is ignoring it and evaluates partially the filter.